portals/knowledge-management/src/Pages/ViewServiceRequest.vue

```diff
@@ -0,0 +1,312 @@
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+-->
+<script setup>
+    import { MagnifyingGlassIcon } from '@heroicons/vue/24/outline';
+    import { defineProps, ref, watch, onMounted } from 'vue';
+    import { useRoute } from 'vue-router';
+    import Breadcrumbs from '../Components/Breadcrumbs.vue';
+    import AppLoading from '../Components/AppLoading.vue';
+    import { consumer } from '../Services/Consumer.js';
+    import { Bars3Icon } from '@heroicons/vue/24/outline/index.js';
+    import { ChevronRightIcon, XMarkIcon } from '@heroicons/vue/20/solid/index.js';
+    import Article from '../Components/Article.vue';
+    import SearchResults from '../Components/SearchResults.vue';
+    import Badge from '../Components/Badge.vue';
+
+    const route = useRoute();
+
+    const props = defineProps({
+        searchUrl: {
+            type: String,
+            required: true,
+        },
+        apiUrl: {
+            type: String,
+            required: true,
+        },
+        serviceRequest: {
+            type: Object,
+            required: true,
+        },
+        serviceRequestUpdates: {
+            type: Array,
+            required: true,
+        },
+    });
+
+    const loadingResults = ref(true);
+    const loadingeSearchResults = ref(true);
+    const serviceRequest = ref(null);
+    const serviceRequestUpdates = ref([]);
+    const searchQuery = ref('');
+    const searchResults = ref(null);
+    const updateMessage = ref('');
+
+    const debounceSearch = debounce((value) => {
+        const { post } = consumer();
+
+        if (!value) {
+            searchQuery.value = null;
+            searchResults.value = null;
+            return;
+        }
+
+        loadingeSearchResults.value = true;
+
+        post(props.searchUrl, {
+            search: JSON.stringify(value),
+        }).then((response) => {
+            searchResults.value = response.data;
+            loadingeSearchResults.value = false;
+        });
+    }, 500);
+
+    watch(searchQuery, (value) => {
+        debounceSearch(value);
+    });
+
+    function debounce(func, delay) {
+        let timerId;
+        return function (...args) {
+            if (timerId) {
+                clearTimeout(timerId);
+            }
+            timerId = setTimeout(() => {
+                func(...args);
+            }, delay);
+        };
+    }
+
+    watch(
+        route,
+        function (newRouteValue) {
+            getData();
+        },
+        {
+            immediate: true,
+        },
+    );
+
+    onMounted(function () {
+        getData();
+    });
```

Comment:
This is not needed due to the watch out the route

Reply:
Done